onnx/backend/test/case/node/__init__.py

```diff
@@ -16,15 +17,67 @@
 _NodeTestCases = []
 _TargetOpType = ""
 
-from onnx.onnx_pb import NodeProto, AttributeProto, TypeProto, FunctionProto
+from onnx.onnx_pb import NodeProto, AttributeProto, TypeProto, FunctionProto, GraphProto
+
+
+def _rename_edges_helper(internal_node: NodeProto,
+                         rename_helper: Callable[[Text], Text],
+                         attribute_map: Dict[Text, AttributeProto],
+                         prefix: Text) -> NodeProto:
+    new_node = NodeProto()
+    new_node.CopyFrom(internal_node)
+    new_node.ClearField("input")
+    new_node.ClearField("output")
+    new_node.ClearField("attribute")
+    for internal_name in internal_node.input:
+        new_node.input.append(rename_helper(internal_name))
+    for internal_name in internal_node.output:
+        new_node.output.append(rename_helper(internal_name))
+    for attr in internal_node.attribute:
+        if attr.HasField("ref_attr_name"):
+            if attr.ref_attr_name in attribute_map:
+                new_attr = AttributeProto()
+                new_attr.CopyFrom(attribute_map[attr.ref_attr_name])  # type: ignore
+                new_attr.name = attr.name
+        else:
+            new_attr = AttributeProto()
+            new_attr.CopyFrom(attr)
+            if attr.type == AttributeProto.GRAPH:
+                new_graph = new_attr.g
+                sg_rename = {}
+                for in_desc in new_graph.input:
+                    sg_rename[in_desc.name] = in_desc.name = prefix + in_desc.name
+                for out_desc in new_graph.output:
+                    sg_rename[out_desc.name] = out_desc.name = prefix + out_desc.name
+                for init_desc in new_graph.initializer:
+                    sg_rename[init_desc.name] = init_desc.name = prefix + init_desc.name
+                for sparse_init_desc in new_graph.sparse_initializer:
+                    sg_rename[sparse_init_desc.values.name] = sparse_init_desc.values.name = prefix + \
+                        sparse_init_desc.values.name
+                for sparse_init_desc in new_graph.sparse_initializer:
+                    sg_rename[sparse_init_desc.indices.name] = sparse_init_desc.indices.name = prefix + \
+                        sparse_init_desc.indices.name
+
+                def subgraph_rename_helper(name: Text) -> Any:
+                    if name in sg_rename:
+                        return sg_rename[name]
+                    else:
+                        return rename_helper(name)
+                new_nodes = [
+                    _rename_edges_helper(node_desc, subgraph_rename_helper, attribute_map, prefix)
+                    for node_desc in new_graph.node
+                ]
+                new_graph.ClearField("node")
+                new_graph.node.extend(new_nodes)
+        new_node.attribute.extend([new_attr])
```

Comment:
I think the above line must be moved into the branches of the if-else statement, like in the original version. Specifically, I think in the else-case of the if-statement in line 38, we should not be inserting new_attr.